test/specs/behaviors/testDefinitions.ts

```diff
@@ -124,20 +124,68 @@ definitions.push({
   },
 })
 
-// Example: Adds attribute 'aria-disabled=true' to 'trigger' component's part based on the property 'disabled'.
+function testMethodConditionallyAddAttribute(
+  parameters,
+  attributeToBeAdded,
+  valueOfAttributeToBeAddedIfTrue,
+  component,
+  propertyDependsOn,
+  valueOfAttributeToBeAddedOtherwise,
+) {
+  const propertyWithAriaSelected = {}
+  const expectedResultAttributeNotDefined = parameters.behavior(propertyWithAriaSelected)
+    .attributes[component][attributeToBeAdded]
+  expect(testHelper.convertToBooleanIfApplicable(expectedResultAttributeNotDefined)).toEqual(
+    testHelper.convertToBooleanIfApplicable(valueOfAttributeToBeAddedOtherwise),
+  )
+
+  propertyWithAriaSelected[propertyDependsOn] = true
+  const expectedResultAttributeDefined = parameters.behavior(propertyWithAriaSelected).attributes[
+    component
+  ][attributeToBeAdded]
+  expect(testHelper.convertToBooleanIfApplicable(expectedResultAttributeDefined)).toEqual(
+    testHelper.convertToBooleanIfApplicable(valueOfAttributeToBeAddedIfTrue),
+  )
+}
+
+// Example: Adds attribute 'aria-disabled=true' to 'trigger' component's part if 'disabled' property is true. Does not set the attribute otherwise.
 definitions.push({
-  regexp: /Adds attribute '([a-z A-Z -]+)=([a-z 0-9]+)' to '([a-z -]+)' component's part based on the property '([a-z -]+)'\./g,
+  regexp: /Adds attribute '([a-zA-Z-]+)=([a-z0-9]+)' to '([a-z-]+)' component's part if '([a-z -]+)' property is true\. Does not set the attribute otherwise\./g,
```

Comment:
👍

Reply:
there are more to fix, will do that in a separate PR.